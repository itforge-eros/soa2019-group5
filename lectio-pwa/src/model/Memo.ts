/**
 * Memo
 * @class
 */
class Memo {
	public id: string;
	public name: string;
	public content: string;
	public audioId: string;
	public tags: Array<any>;

	constructor(id: string, name: string, content: string, audioId: string, tags: Array<any>) {
		this.id = id;
		this.name = name;
		this.content = content;
		this.audioId = audioId;
		this.tags = tags;
	}
}

export default Memo;
