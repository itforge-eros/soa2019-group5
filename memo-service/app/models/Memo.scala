package models

import java.time.LocalDateTime
import java.util.UUID

case class Memo(
                 id: UUID = UUID.randomUUID,
                 title: String = "",
                 content: String = "",
                 summary: String = "",
                 tags: Set[String] = Set.empty,
                 created_time: LocalDateTime = LocalDateTime.now(),
                 updated_time: LocalDateTime = LocalDateTime.now()
               )
