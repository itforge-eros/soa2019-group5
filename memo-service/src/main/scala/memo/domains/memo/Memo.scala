package memo.domains.memo

import java.time.Instant
import java.util.UUID

case class Memo(
                 id: String = UUID.randomUUID().toString,
                 title: String = "",
                 content: String = "",
                 summary: String = "",
                 tags: Set[String] = Set.empty,
                 created_time: Instant = Instant.now(),
                 updated_time: Instant = Instant.now()
               )
