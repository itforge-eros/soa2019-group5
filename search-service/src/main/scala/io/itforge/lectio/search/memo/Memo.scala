package io.itforge.lectio.search.memo

import java.util.{Date, UUID}

case class Memo(
    uuid: UUID,
    title: String,
    content: String,
    tags: Set[String],
    user_id: UUID,
    created_time: Date)
