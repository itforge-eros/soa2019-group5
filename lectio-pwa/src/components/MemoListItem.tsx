import React, { Component } from 'react';
import { ListItem, ListItemText } from "@material-ui/core";
import { RouteComponentProps, withRouter } from 'react-router-dom';
import { listWithCommas } from '../utils/fmt';
import Memo from '../model/Memo';

type theProp = {
	memo: Memo | serverMemo,
	history: any,
	selected?: boolean,
	schema: 'local' | 'server'
};

/**
 * A ListItem for displaying a Memo object
 */
class MemoListItem extends Component<theProp & RouteComponentProps<{}>, any> {
	constructor(props: any) {
		super(props);
		this.state = {
			memoName: '',
			memoTags: '',
			memoId: ''
		};
	}

	componentWillMount(): void {
		const isLocal = this.props.schema === 'local';

		// @ts-ignore
		const memoName = isLocal ? this.props.memo.name : this.props.memo.title;
		// @ts-ignore
		const memoId = isLocal ? this.props.memo.id : this.props.memo.uuid;
		const memoTags: Array<string> = isLocal ?
			// @ts-ignore
			this.props.memo.tags.map((t: MemoTag) => t.name) :
			this.props.memo.tags;

		this.setState({ memoName, memoId, memoTags });
	}

	handleItemClick() {
		const url = '/memo/' + this.state.memoId;
		setTimeout(() => this.props.history.push(url), 200);
	}

  render() {
		return (
			<ListItem
				button divider
				selected={this.props.selected}
				onClick={() => this.handleItemClick()}
				>
				<ListItemText
					primary={this.state.memoName}
					secondary={listWithCommas(this.state.memoTags)} />
			</ListItem>
		)
	}
}

export default withRouter(MemoListItem);
