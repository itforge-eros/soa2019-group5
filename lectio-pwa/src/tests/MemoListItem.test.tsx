import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import MemoListItem from "../components/MemoListItem";
import Memo from "../model/Memo";

const localMemo: Memo = new Memo(
	'sample-id',
	'Lorem Ipsum',
	'This is the content',
	'sample-id',
	[
		{id: 'tag-id-1', name: 'Tag 1'},
		{id: 'tag-id-2', name: 'Tag 2'}
		]
);

const serverMemo: serverMemo = {
	uuid: 'sample-id',
	title: 'Lorem Ipsum',
	content: 'This is the content',
	tags: ['Tag 1', 'Tag 2']
};

it('renders MemoListItem with server schema successfully', () => {
	const tree = renderer
		.create(
			<BrowserRouter>
				<MemoListItem memo={serverMemo} selected={false} schema="server"/>
			</BrowserRouter>
		)
		.toJSON();
	expect(tree).toMatchSnapshot();
});

it('renders MemoListItem with local schema successfully', () => {
	const tree = renderer
		.create(
			<BrowserRouter>
				<MemoListItem memo={localMemo} selected={false} schema="local"/>
			</BrowserRouter>
		)
		.toJSON();
	expect(tree).toMatchSnapshot();
});

export default undefined;
