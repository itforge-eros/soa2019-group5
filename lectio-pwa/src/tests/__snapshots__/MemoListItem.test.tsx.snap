// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MemoListItem with local schema successfully 1`] = `
<div
  className="MuiButtonBase-root-13 MuiListItem-root-1 MuiListItem-default-4 MuiListItem-gutters-9 MuiListItem-divider-8 MuiListItem-button-10"
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="button"
  tabIndex="0"
>
  <div
    className="MuiListItemText-root-16"
  >
    <span
      className="MuiTypography-root-22 MuiTypography-subheading-29 MuiListItemText-primary-19"
    >
      Lorem Ipsum
    </span>
    <p
      className="MuiTypography-root-22 MuiTypography-body1-31 MuiTypography-colorTextSecondary-55 MuiListItemText-secondary-20"
    >
      Tag 1, Tag 2
    </p>
  </div>
  <span
    className="MuiTouchRipple-root-58"
  />
</div>
`;

exports[`renders MemoListItem with server schema successfully 1`] = `
<div
  className="MuiButtonBase-root-13 MuiListItem-root-1 MuiListItem-default-4 MuiListItem-gutters-9 MuiListItem-divider-8 MuiListItem-button-10"
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="button"
  tabIndex="0"
>
  <div
    className="MuiListItemText-root-16"
  >
    <span
      className="MuiTypography-root-22 MuiTypography-subheading-29 MuiListItemText-primary-19"
    >
      Lorem Ipsum
    </span>
    <p
      className="MuiTypography-root-22 MuiTypography-body1-31 MuiTypography-colorTextSecondary-55 MuiListItemText-secondary-20"
    >
      Tag 1, Tag 2
    </p>
  </div>
  <span
    className="MuiTouchRipple-root-58"
  />
</div>
`;
